pyramid_jsonapi/__init__.py

```diff
@@ -184,6 +181,27 @@ def create_jsonapi(self, engine=None, test_data=None, api_version=''):
         if self.settings.metadata_endpoints:
             self.metadata = pyramid_jsonapi.metadata.MetaData(self)
 
+        # Add error views
+        prefnames = ['api']
+        if self.settings.metadata_endpoints:
+            prefnames.append('metadata')
+        for prefname in prefnames:
+            setting_name = 'route_pattern_{}_prefix'.format(prefname)
+            path_info = "/"
+            setting = getattr(self.settings, setting_name)
```

Comment:
This could maybe be neater as `str(getattr(...))` as then the test for content can just be `if setting:` ?

Reply:
If the setting is ever `None` then the string representation of that is `'None'` which evaluates to `True` - so not the same logic. Can those settings ever be `None`?

Reply:
The `config_defaults` dict in `__init__.py` sets these to `api` and `metadata` - you can't write `None` via an inifile - and if you did force it to `None` with some hacking the code would almost certainly break in many other places!